c-wrapper/c-examples/challenge_response/Makefile

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 Contributors to the Veraison project.
+# SPDX-License-Identifier: Apache-2.0
+
+default: run
+
+# If the header has not been generated, then we need to 'cargo build' the root workspace
+../../veraison_client_wrapper.h:
+	pushd ../../.. && cargo build && popd
+
+challenge_response.o: challenge_response.c ../../veraison_client_wrapper.h
+	gcc -Wall -c -I../.. challenge_response.c
```

Comment:
Ah, of course. Thanks for reminding me. It has been quite a while since I wrote a Makefile from scratch, and this one was templated on a similar C example program from within the Parsec project. That's a simple enough change, so I can tidy that up before we merge.